### PR TITLE
 Add license detection rules and test for badge URLs #1380

### PR DESCRIPTION
**Pull request by mriiinal**
This adds support for #1380 - license in badge URL format

Reported-by: Philippe Ombredanne <pombredanne@nexb.com> 
Signed-off-by: Mrinal Paliwal <mrinal16164@iiitd.ac.in>

### CODE REVIEW

**Comment by pombredanne** [comment]
@mriiinal Thank you. Can you add your sign off to your commit messages? (or at least to the PR message for this one time?) See https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst for details

File: tests/licensedcode/data/licenses/url_badge.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+license_expressions:
+  -agpl-3.0
```
Comment (pombredanne) [inline tests/licensedcode/data/licenses/url_badge.yml]: you likely want a space after each dash as in 
`    - agpl-3.0`

**Comment by mriiinal** [comment]
@pombredanne Aplogies, I've updated the PR message accordingly. I've signed off my commit messages. Please let me know I'm doing it incorrectly, I'm new to this.
image

File: tests/licensedcode/data/licenses/url_badge.yml (hunk 1 above)
Comment (mriiinal) [reply tests/licensedcode/data/licenses/url_badge.yml]: - Yes, the scan is detecting the new rules
- Fixed indentation of url_badge.yml

**Comment by mriiinal** [comment]
@pombredanne I fixed the test files, I think this is complete now. Let me know if there are any changes

**Comment by pombredanne** [comment]
All good. Thank you ++

**Pull request merged**